rlclientlib/extensions/onnx/src/onnx_model.cc

```diff
@@ -0,0 +1,204 @@
+#include <memory>
+#include <sstream>
+#include <assert.h>
+
+#include "tensor_notation.h"
+
+#include "trace_logger.h"
+#include "err_constants.h"
+#include "onnx_model.h"
+#include "str_util.h"
+#include "api_status.h"
+
+#include "factory_resolver.h"
+
+// for base64 decoding
+#include <cpprest/http_client.h>
+
+namespace reinforcement_learning { namespace onnx {
+
+  inline void OrtLogCallback(
+    void* param, OrtLoggingLevel severity, const char* category, const char* logid, const char* code_location,
+    const char* message)
+  {
+    i_trace* trace_logger = static_cast<i_trace*>(param);
+
+    int loglevel = LEVEL_ERROR;
+    switch (severity)
+    {
+      case ORT_LOGGING_LEVEL_VERBOSE:
+        loglevel = LEVEL_DEBUG;
+      break;
+      
+      case ORT_LOGGING_LEVEL_INFO:
+        loglevel = LEVEL_INFO;
+      break;
+
+      case ORT_LOGGING_LEVEL_WARNING:
+        loglevel = LEVEL_WARN;
+      break;
+
+      case ORT_LOGGING_LEVEL_FATAL: // TODO: Should this be a background error?
+      case ORT_LOGGING_LEVEL_ERROR:
+        loglevel = LEVEL_ERROR;
+      break;
+    }
+
+    std::stringstream buf;
+    buf << "[onnxruntime, modelid=" << logid << "]: " << message;
+
+    TRACE_LOG(trace_logger, loglevel, buf.str());
```

Comment:
If trace log is not being used we should not pay the cost fo the stringstream above